peachjam/js/components/DocDiffs/DocumentDiffDropdown.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div class="d-flex">
+    <div class="dropdown ml-3">
+      <button
+        class="btn btn-sm dropdown-toggle"
+        :class="btnClass"
+        type="button"
+        data-toggle="dropdown"
+        aria-haspopup="true"
+        aria-expanded="false"
+      >
+        Compare to
+        <span v-if="compareTo">{{ compareTo.date }}</span>
+      </button>
+      <div
+        class="dropdown-menu"
+        aria-labelledby="dropdownMenuButton"
+      >
+        <a
+          v-for="doc in documents"
+          :key="doc.id"
+          class="dropdown-item"
+          :class="itemClass(doc)"
+          href="#"
+          @click.prevent="toggle(doc)"
+        >
+          {{ doc.date }}
+        </a>
+      </div>
+    </div>
+    <div
+      v-if="loading"
+      class="ml-3"
+    >
+      Loading...
+    </div>
+    <div
+      v-if="compareTo"
+      class="ml-2"
+    >
+      <label>
+        <input
+          v-model="sideBySide"
+          type="checkbox"
+        > Show changes side-by-side
+      </label>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'DocumentDiffDropdown',
+  props: ['documents', 'aknRoot', 'documentId'],
+
+  data: () => ({
+    compareTo: null,
+    loading: false,
+    sideBySide: true
+  }),
+
+  computed: {
+    btnClass () {
+      return this.compareTo === null ? 'btn-outline-secondary' : 'btn-primary';
+    }
+  },
+
+  watch: {
+    compareTo () {
+      this.displayDiff();
+    },
+
+    sideBySide () {
+      this.displayDiff();
+    }
+  },
+
+  mounted () {
+    this.cleanAknHtml = this.aknRoot.innerHTML;
+  },
+
+  methods: {
+    toggle (doc) {
+      if (this.compareTo === doc) {
+        this.compareTo = null;
+      } else {
+        this.compareTo = doc;
+      }
+    },
+
+    itemClass (doc) {
+      return doc === this.compareTo ? 'active' : '';
+    },
+
+    async displayDiff () {
+      if (this.compareTo) {
+        this.loading = true;
+        const resp = await fetch(`/reader/${this.documentId}/document-diff?other=${this.compareTo.id}`);
```

Comment:
This must be the services.lawsafrica.com URL, using the current host as the partner code. (eg. `.../p/${document.location.host}/...`

Reply:
Maybe provide a helper function to build this URL, because you need it in two places and for debugging purposes, it would actually be easier to determine the partner code as: use `laws.africa` if the current host is `localhost`, otherwise use the current host.

That way, when developing locally we use `laws.africa` which allows the `localhost` referrer, but in production it uses `senlii.org` or whatever.

Reply:
Ah, crap, I commented in the wrong place.

This whole file can go, we don't use it here at all. But the comment still applies to the URL elsewhere :)

Reply:
done